scripts/delete_from_repository.py

```diff
@@ -0,0 +1,144 @@
+    parser.add_argument('-p', '--packages', nargs='+', required=True, type=str)
+    parser.add_argument('-v', '--verbose', action='store_true', default=False)
+    return parser.parse_args(args)
+
+
+def get_repository(repo_name: str, arch: str) -> models.Repository:
+    with SyncSession() as session:
+        repo = (session.execute(select(models.Repository)
+                                .where(models.Repository.name == repo_name,
+                                       models.Repository.arch == arch))
+                .scalars().first())
+        if not repo:
+            raise RepositoryNotFound(f'Repository {repo_name} is not found')
+        return repo
+
+
+def wait_for_task(base_url: str, task_href: str, auth: HTTPBasicAuth):
+    task_url = urljoin(base_url, task_href)
+    res = requests.get(task_url, auth=auth).json()
+    while res['state'] not in ('failed', 'completed'):
+        time.sleep(3)
+        res = requests.get(task_url, auth=auth).json()
+    return res
+
+
+def publish_repo(base_url: str, repo_href: str, auth: HTTPBasicAuth):
+    full_url = urljoin(base_url, '/pulp/api/v3/publications/rpm/rpm/')
+    task_href = requests.post(
+        full_url, json={'repository': repo_href}, auth=auth).json()['task']
+    return wait_for_task(base_url, task_href, auth)
+
+
+def setup_logging(verbose: bool = False):
+    log_level = logging.DEBUG if verbose else logging.INFO
+    logging.basicConfig(format='%(asctime)s %(levelname)-8s %(message)s',
+                        level=log_level, datefmt='%Y-%m-%d %H:%M:%S')
+
+
+def main():
+    packages_endpoint = '/pulp/api/v3/content/rpm/packages/'
+    arguments = parse_args(sys.argv[1:])
+    setup_logging(verbose=arguments.verbose)
+    logger = logging.getLogger(PROG_NAME)
+    pulp_auth = HTTPBasicAuth(
+        settings.pulp_user, settings.pulp_password)
+    pulp_host = settings.pulp_host
+
+    repo = get_repository(
+        arguments.repository_name, arguments.architecture)
+    logger.debug('Repository: %s', repo)
+    repo_info = requests.get(
+        urljoin(pulp_host, repo.pulp_href), auth=pulp_auth).json()
+    # Looking for packages that belong to the latest repository version
+    query_params = {
+        'repository_version': repo_info['latest_version_href'],
+        'fields': ['location_href', 'pulp_href'],
+    }
+    # Check how many packages is in the repo to get them the in 1 request
+    logger.info('Getting repository packages')
+    pulp_response = requests.get(
+        urljoin(pulp_host, packages_endpoint), params=query_params,
+        auth=pulp_auth).json()
+    package_count = pulp_response['count']
+    repo_packages = pulp_response['results']
+    logger.debug('Packages count: %d', package_count)
+    if package_count > 100:
+        query_params['limit'] = package_count + 1
+        repo_packages = requests.get(
+            urljoin(pulp_host, packages_endpoint),
+            params=query_params, auth=pulp_auth).json()['results']
```

Comment:
we rewrite here results from prev responses, we should extend list instead
also, increasing limit by one is kinda strange, it seems like slow us down
u can look on `iter_repo` func in `alws/utils/uploader.py:18` how we get all content from repo